src/preferences.h

```diff
@@ -17,180 +17,101 @@
 #define PREFERENCES_H_
 
+
 #include <sys/keycodes.h>
 #include <libconfig.h>
+#include <unicode/utf.h>
+#include <errno.h>
+
+#include "types.h"
+#include "symmenu.h"
 #include "SDL.h"
 
+#define PREFS_FILE_PATH ".term48rc"
+#define README_FILE_PATH "../app/native/README"
+#define README45_FILE_PATH "../app/native/README45"
+
 #define PREFS_COLOR_NUM_ELEMENTS 3
 #define PREFS_SYMKEYS_DEFAULT_NUM_ROWS 2
 
-#define PREFS_V6_KDEF_K "\x1b[A"
-#define PREFS_V6_KDEF_J "\x1b[B"
-#define PREFS_V6_KDEF_L "\x1b[C"
-#define PREFS_V6_KDEF_H "\x1b[D"
-#define PREFS_V7_KDEF_K "kcuu1"
-#define PREFS_V7_KDEF_J "kcud1"
-#define PREFS_V7_KDEF_L "kcuf1"
-#define PREFS_V7_KDEF_H "kcub1"
-
-static int PREFS_VERSION = 8;
+static int PREFS_VERSION = 9;
 
-struct symkey_entry {
-	const char* name;
-	const char* c;
-  char flash;
-	UChar* uc;
-	int x, y, off_x, off_y;
-	SDL_Surface* background;
-	SDL_Surface* symbol;
-	SDL_Surface* flash_symbol;
-	SDL_Surface* key;
-};
+#define DEFAULT_FONT_PATH "/usr/fonts/font_repository/monotype/cour.ttf"
+#define DEFAULT_FONT_SIZE 24//40
+#define DEFAULT_TEXT_COLOR (int[]){255, 255, 255}
+#define DEFAULT_BACKGROUND_COLOR (int[]){0, 0, 0}
+#define DEFAULT_SCREEN_IDLE_AWAKE 0
+#define DEFAULT_AUTO_SHOW_VKB 1
+#define DEFAULT_METAMODE_DOUBLETAP_KEY KEYCODE_RIGHT_SHIFT
+#define DEFAULT_METAMODE_DOUBLETAP_DELAY 500000000
+#define DEFAULT_KEYHOLD_ACTIONS 1
+#define DEFAULT_METAMODE_HOLD_KEY KEYCODE_SPACE
+#define DEFAULT_ALLOW_RESIZE_COLUMNS 0
+#define DEFAULT_METAMODE_HITBOX (int[]){0, 0, 100, 100}
+#define DEFAULT_TTY_ENCODING "UTF-8"
+#define DEFAULT_METAMODE_KEYS_LEN 2
+#define DEFAULT_METAMODE_KEYS (keymap_t[]){{'e', "\x1b"}, {'t', "\x09"}}
+#define DEFAULT_METAMODE_STICKY_KEYS_LEN 4
+#define DEFAULT_METAMODE_STICKY_KEYS (keymap_t[]){{'k', "kcuu1"}, \
+                                                  {'j', "kcud1"}, \
+                                                  {'l', "kcuf1"}, \
+                                                  {'h', "kcub1"}}
+#define DEFAULT_METAMODE_FUNC_KEYS_LEN 4
+#define DEFAULT_METAMODE_FUNC_KEYS (keymap_t[]){{'a', "alt_down"}, \
+                                                {'c', "ctrl_down"}, \
+                                                {'s', "rescreen"}, \
+                                                {'v', "paste_clipboard"}}
+#define DEFAULT_SYMMENU_NUM_ROWS 2
+#define DEFAULT_SYMMENU_ROW_LENS (int[]){10, 9}
+#define DEFAULT_SYMMENU_ENTRIES (keymap_t[]) {  \
+    {'q', "~"}, {'w', "`"}, {'e', "{"}, {'r', "}"}, {'t', "["}, {'y', "]"}, {'u', "<"}, {'i', ">"}, {'o', "^"}, {'p', "%"}, \
+    {'a', "="}, {'s', "-"}, {'d', "*"}, {'f', "/"}, {'g', "\\"},{'h', "|"}, {'j', "&"}, {'k', "'"}, {'l', "\""} \
+}
+#define DEFAULT_STICKY_SYM_KEY 0
+#define DEFAULT_STICKY_SHIFT_KEY 0
+#define DEFAULT_STICKY_ALT_KEY 0
+#define DEFAULT_KEYHOLD_ACTIONS_EXEMPT_LEN 2
+#define DEFAULT_KEYHOLD_ACTIONS_EXEMPT (int[]){KEYCODE_BACKSPACE, KEYCODE_RETURN}
+#define DEFAULT_RESCREEN_FOR_SYMMENU 1
```

Comment:
This is to avoid resizing the terminal window when the symmenu comes up, yes? Cool. It should be documented in the README and README45.